### PR TITLE
Refactor: Single-Module Wrappers By Default

### PR DESCRIPTION
**Pull request by dOrgJelli**
closes: https://github.com/polywrap/monorepo/issues/886, https://github.com/polywrap/monorepo/issues/887, https://github.com/polywrap/monorepo/issues/888, https://github.com/polywrap/monorepo/issues/889

# Features
- The concept of `Mutation` and `Query` modules has been deprecated. Now all wrapper methods are defined within a single `Module`.

# Breaking Changes
- All wrappers created prior to this change are incompatible.

### CODE REVIEW

File: packages/cli/src/lib/SchemaComposer.ts [hunk 1]
```diff
@@ -64,17 +64,13 @@ export class SchemaComposer {
...
-        throw Error(`Schema "${name}" cannot be loaded at path: ${schemaPath}`);
-      }
-
-      options.schemas[name] = schemaFile;
+    const schemaFile = getSchemaFile(schemaNamedPath);
+    if (!schemaFile) {
+      throw Error(`Schema cannot be loaded at path: ${schemaNamedPath}`);
     }
 
+    options.schemas["main"] = schemaFile;
```
Comment (cbrzn) [inline packages/cli/src/lib/SchemaComposer.ts]: Note: This should change to just schema - But left it that way (for now) to make it compliant with `composeSchema` interface

File: packages/cli/src/lib/project/PluginProject.ts [hunk 2]
```diff
@@ -121,73 +107,31 @@ export class PluginProject extends Project<PluginManifest> {
...
-
+    resetDir(moduleDirectory);
     const bindLanguage = pluginManifestLanguageToBindLanguage(
       await this.getManifestLanguage()
     );
 
     const options: BindOptions = {
       projectName: manifest.name,
-      modules: [],
+      modules: [
```
Comment (cbrzn) [inline packages/cli/src/lib/project/PluginProject.ts]: Same here, should be `module` and be an object instead of an array of objects

File: packages/js/core/src/types/Env.ts [hunk 3]
```diff
@@ -7,13 +7,7 @@ export interface Env<TUri = string> {
   uri: TUri;
 
   /** Env variables shared by both mutation and query */
```
Comment (cbrzn) [inline packages/js/core/src/types/Env.ts]: Woops, this should be changed

**Comment by cbrzn** [comment]
lmao @ the lines of code being removed here :fire:

File: packages/cli/src/lib/Compiler.ts [hunk 4]
```diff
@@ -499,79 +411,26 @@ export class Compiler {
...
 
   private _validateState(state: CompilerState) {
-    const { composerOutput, modulesToBuild, web3ApiManifest } = state;
+    const { composerOutput, web3ApiManifest } = state;
 
-    const throwMissingSchema = (moduleName: string) => {
-      const missingSchemaMessage = intlMsg.lib_compiler_missingSchema({
-        name: `"${moduleName}"`,
-      });
+    if (!composerOutput.main || !composerOutput.main.schema) {
```
Comment (cbrzn) [inline packages/cli/src/lib/Compiler.ts]: Should this be `composerOutput.module` ?

File: packages/cli/src/lib/SchemaComposer.ts (hunk 1 above)
Comment (cbrzn) [inline packages/cli/src/lib/SchemaComposer.ts]: Should this be changed to `options.schemas["module"]` ? Probably related to my previous comment :thinking:

File: packages/cli/src/lib/project/PluginProject.ts [hunk 5]
```diff
@@ -124,73 +110,26 @@ export class PluginProject extends Project<PluginManifest> {
...
-
+    resetDir(moduleDirectory);
     const bindLanguage = pluginManifestLanguageToBindLanguage(
       await this.getManifestLanguage()
     );
 
     const options: BindOptions = {
       projectName: manifest.name,
-      modules: [],
+      typeInfo: composerOutput.main?.typeInfo as TypeInfo,
```
Comment (cbrzn) [inline packages/cli/src/lib/project/PluginProject.ts]: Same here - Should we move to `module`? I think that all of these comments (the ones that are `main` -> `module`) are related. Let me know if this makes sense and I can tackle this

File: packages/cli/src/lib/project/Web3ApiProject.ts [hunk 6]
```diff
@@ -165,69 +153,27 @@ export class Web3ApiProject extends Project<Web3ApiManifest> {
...
+    resetDir(moduleDirectory);
 
     const bindLanguage = web3apiManifestLanguageToBindLanguage(
       await this.getManifestLanguage()
     );
 
     const options: BindOptions = {
       projectName: manifest.name,
-      modules: [],
+      typeInfo: composerOutput.main?.typeInfo as TypeInfo,
```
Comment (cbrzn) [inline packages/cli/src/lib/project/Web3ApiProject.ts]: `main` -> `module` ?

File: packages/js/core/src/__tests__/manifest/web3api/incorrect-version-format/web3api.yaml [hunk 7]
```diff
@@ -1,13 +1,3 @@
...
-  module:
-    language: wasm/assemblyscript
-    file: ../query.ts
-query:
-  schema:
-    file: ../schema.graphql
-  module:
-    language: wasm/assemblyscript
-    file: ../mutation.ts
+main: ../index.ts
```
Comment (cbrzn) [inline packages/js/core/src/__tests__/manifest/web3api/incorrect-version-format/web3api.yaml]: I guess the format validation comes first and that's why it's not failing but this should be changed to `module`

File: packages/js/core/src/__tests__/manifest/web3api/required-field-missing/web3api.yaml [hunk 8]
```diff
@@ -1,12 +1,4 @@
...
-query:
-  schema:
-    file: ../schema.graphql
-  module:
-    language: wasm/assemblyscript
-    file: ../mutation.ts
+format: 0.0.1-prealpha.9
+name: "Required missing field"
+language: wasm/assemblyscript
+main: ../index.ts
```
Comment (cbrzn) [inline packages/js/core/src/__tests__/manifest/web3api/required-field-missing/web3api.yaml]: `main` -> `module` again

File: packages/js/core/src/__tests__/resolveUri.spec.ts [hunk 9]
```diff
@@ -153,57 +152,52 @@ describe("resolveUri", () => {
         options: GetManifestOptions<TManifestType>,
         client: Client
       ) => {
-        const manifest = {
-          format: "0.0.1-prealpha.8",
+        const manifest: unknown = {
+          format: "0.0.1-prealpha.9",
           language: "",
-          modules: {},
+          main: "",
```
Comment (cbrzn) [inline packages/js/core/src/__tests__/resolveUri.spec.ts]: lol I am being annoying with `main` -> `module` but I am just commenting this so I can know where to go based on comments

File: packages/schema/compose/src/templates/schema.mustache.ts [hunk 10]
```diff
@@ -1,6 +1,6 @@
 const template = `
```
Comment (cbrzn) [inline packages/schema/compose/src/templates/schema.mustache.ts:1]: We need to rename `Queries` to `Modules` here

File: packages/cli/src/__tests__/e2e/deploy.spec.ts [hunk 11]
```diff
@@ -37,6 +37,9 @@ const setup = async (domainNames: string[]) => {
   await stopTestEnvironment();
   await initTestEnvironment();
 
+  // Wait a little longer just in case
+  await new Promise((resolve) => setTimeout(resolve, 3000));
+
```
Comment (namesty) [inline packages/cli/src/__tests__/e2e/deploy.spec.ts:42]: Why is this necessary?

File: packages/test-cases/cases/apis/wasm-rs/bigint-type/web3api.yaml [hunk 12]
```diff
@@ -1,7 +1,6 @@
...
+format: 0.0.1-prealpha.9
+name: BigInt
 language: wasm/rust
 build: ./web3api.build.yaml
-modules:
-  query:
-    schema: ./query/schema.graphql
-    module: ./query/Cargo.toml
+schema: ./schema.graphql
+module: ./Cargo.toml
```
Comment (cbrzn) [inline packages/test-cases/cases/apis/wasm-rs/bigint-type/web3api.yaml:6]: Probably out of place but shouldn't this point to a `.rs` file? Wouldn't this be the same as pointing it to ` package.json` ?

File: packages/test-cases/cases/apis/wasm-rs/bignumber-type/web3api.yaml [hunk 13]
```diff
@@ -1,7 +1,6 @@
...
+format: 0.0.1-prealpha.9
+name: BigNumber
 language: wasm/rust
 build: ./web3api.build.yaml
-modules:
-  query:
-    schema: ./query/schema.graphql
-    module: ./query/Cargo.toml
+schema: ./schema.graphql
+module: ./Cargo.toml
```
Comment (cbrzn) [inline packages/test-cases/cases/apis/wasm-rs/bignumber-type/web3api.yaml:6]: Same as previous comment

File: packages/test-cases/cases/cli/plugin/codegen/002-single-module/expected/src/w3/schema.ts [hunk 14]
```diff
@@ -181,6 +183,50 @@ type Ethereum_Query @imported(
...
+  ): String!
+
+  sendRPC(
+    method: String!
+    params: [String!]!
+    connection: Ethereum_Connection
+  ): String
 }
 
 ### Imported Queries END ###
```
Comment (cbrzn) [inline packages/test-cases/cases/cli/plugin/codegen/002-single-module/expected/src/w3/schema.ts]: Once the template file is changed, I guess this line should change?

File: packages/test-cases/cases/cli/plugin/codegen/004-env-sanitization/expected/src/w3/schema.ts [hunk 15]
```diff
@@ -38,26 +38,24 @@ directive @annotate(type: String!) on FIELD
...
-type QueryEnv {
+type Env {
   queryArg: String!
 }
 
-type MutationEnv {
-  mutationArg: String!
-}
-
 ### Imported Queries START ###
```
Comment (cbrzn) [inline packages/test-cases/cases/cli/plugin/codegen/004-env-sanitization/expected/src/w3/schema.ts]: This should be removed too

**Review by dOrgJelli — COMMENTED** [review]
Just went through everything and it all LGTM!

File: packages/test-cases/cases/apis/wasm-rs/bigint-type/web3api.yaml (hunk 12 above)
Comment (dOrgJelli) [reply packages/test-cases/cases/apis/wasm-rs/bigint-type/web3api.yaml:6]: Talked privately about this, going to take note of this and revisit v soon

File: packages/js/core/src/__tests__/resolveUri.spec.ts [hunk 16]
```diff
@@ -231,10 +225,10 @@ describe("resolveUri", () => {
...
   ];
 
-  const apis: Record<string, PluginModules> = {
-    "w3://ens/ens": ensApi,
-    "w3://ens/ipfs": ipfsApi,
-    "w3://ens/my-plugin": pluginApi,
+  const apis: Record<string, PluginModule> = {
+    "w3://ens/ens": ensApi as unknown as PluginModule,
+    "w3://ens/ipfs": ipfsApi as unknown as PluginModule,
+    "w3://ens/my-plugin": pluginApi as unknown as PluginModule,
```
Comment (namesty) [inline packages/js/core/src/__tests__/resolveUri.spec.ts:231]: If we need to do this kind of casting, wouldn't it make sense to have an interface that abstracts part of what currently is in the abstract class?

**Review by namesty — APPROVED** [review]

File: packages/js/core/src/__tests__/resolveUri.spec.ts (hunk 16 above)
Comment (dOrgJelli) [reply packages/js/core/src/__tests__/resolveUri.spec.ts:231]: this is indeed a dirty hack, and should be cleaned up in the future, but nothing "wrong" with it right now

File: packages/cli/src/__tests__/e2e/deploy.spec.ts (hunk 11 above)
Comment (dOrgJelli) [reply packages/cli/src/__tests__/e2e/deploy.spec.ts:42]: This test was inconsistently passing, and this fixed it. Not entirely sure why, maybe it's the new ENS setup step not finishing quick enough?
